### ISSUE TITLE
Add Gitlab as support Container Registry

### ISSUE THREAD

**Issue by gwright99** [seqeralabs/nf-tower-docs#401]
As per [FL86 3738](https://support.seqera.io/a/tickets/3738), Wave will work with container manifests stored in Gitlab.

This should be reflected in our documentation. I would be worthwhile to also add other known working solutions like Gitea and GitHub Packages.

**Comment by jason-seqera** [comment]
Could possible list supported providers here? https://help.tower.nf/23.1/credentials/overview/

**Comment by gwright99** [comment]
> Could possible list supported providers here? https://help.tower.nf/23.1/credentials/overview/

Yeah, that's the section - specifically **container registry credentials**.

![image](https://github.com/seqeralabs/nf-tower-docs/assets/1309341/7046d720-2c96-4360-8cd9-e2761df9e8f3)

**Comment by jason-seqera** [comment]
@gwright99, do you know who I could ask to elaborate on all the supported container registries, or is it GitLab, Gitea, and GitHub Packages?

Thanks!

**Comment by gwright99** [comment]
The general process of calling a container registry seems to be the same so that means we should be able to support most of the solutions out there. 

There are gotchas occasionally (e.g. JFrog) where the way the layer was returned was different than Wave expected. You'll need to ask Engineering for more details re: the edge case - to my knowledge, it was never super clearly explained why JFrog did differently (beyond "it returned the data itself, not a redirect link) so I don't know how/if this same sort of problem could re-express itself in future with another solution.

That caveat aside, we know GitHub packages, GitLab, and Gitea are all supported (and proven working solutions) so this would be a quick update win while the other discussion occurs in parallel.

**Comment by pditommaso** [comment]
True, they all implement Docker registry v2 API. However, most of the time, each of them has its own messed authenticate security/credentials mechanics.

Explaining how to get those keys and to use in Tower is very useful
